src/Sylius/Component/Translation/Provider/CurrentLocaleProvider.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Translation\Provider;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+class CurrentLocaleProvider implements CurrentLocaleProviderInterface
```

Comment:
In PR description, I said:

Things left to do:

Make all scenarios green
Fix few specs
Allow same thing in SyliusResourceBundle resources configuration
Rename CurrentLocaleProviderInterface to LocaleProviderInterface
Update MongoDB driver